Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -197,92 +213,18 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
-      Signal.combineLatest(self.configureWithDataProperty.signal.skipNil(), checkoutId)
-        .takeWhen(self.applePayButtonTappedSignal)
-        .switchMap { initialData, checkoutId in
-          let projectId = initialData.project.graphID
-          let pledgeTotal = initialData.total
-
-          return stripeIntentService.createPaymentIntent(
-            for: projectId,
-            checkoutId: checkoutId,
-            pledgeTotal: pledgeTotal
-          )
-          .materialize()
-        }
-
-    let newPaymentIntentForApplePayError: Signal<ErrorEnvelope, Never> = createPaymentIntentForApplePay
-      .errors()
-
-    let newPaymentIntentForApplePay: Signal<String, Never> = createPaymentIntentForApplePay
-      .values()
-      .map { $0.clientSecret }
-
-    self.goToApplePayPaymentAuthorization = self
-      .configureWithDataProperty
-      .signal
-      .skipNil()
-      .combineLatest(with: newPaymentIntentForApplePay)
-      .map { (
-        data: PostCampaignCheckoutData,
-        paymentIntent: String
-      ) -> PostCampaignPaymentAuthorizationData? in
-        let baseReward = data.baseReward
-
-        return PostCampaignPaymentAuthorizationData(
-          project: data.project,
-          hasNoReward: baseReward.isNoReward,
-          subtotal: baseReward.isNoReward ? baseReward.minimum : calculateAllRewardsTotal(
-            addOnRewards: data.rewards,
-            selectedQuantities: data.selectedQuantities
-          ),
-          bonus: data.bonusAmount ?? 0,
-          shipping: data.shipping?.total ?? 0,
-          total: data.total,
-          merchantIdentifier: Secrets.ApplePay.merchantIdentifier,
-          paymentIntent: paymentIntent
-        )
-      }
-      .skipNil()
+    /// See ApplePayCheckoutViewModel.swift
 
-    let validateCheckoutWithApplePay = Signal.combineLatest(
-      newPaymentIntentForApplePay,
-      checkoutId,
-      self.applePayPaymentMethodIdSignal.signal
+    let applePayErrors = Signal.merge(
+      self.applePayViewModel.createNewPaymentIntentError,
+      self.applePayViewModel.validateCheckoutError
     )
-    .takeWhen(self.applePayContextDidCompleteSignal)
-    .switchMap { (clientSecret: String, checkoutId: String, paymentMethodId: String) in
-      AppEnvironment.current.apiService
-        .validateCheckout(
-          checkoutId: checkoutId,
-          paymentSourceId: paymentMethodId,
-          paymentIntentClientSecret: clientSecret
-        )
-        .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
-        .materialize()
-    }
 
-    let validateCheckoutError = Signal
-      .merge(
-        validateCheckoutExistingCard.errors(),
-        validateCheckoutWithApplePay.errors()
-      )
+    let validateCheckoutExistingCardError = validateCheckoutExistingCard.errors()
 
-    self.showErrorBannerWithMessage = validateCheckoutError
+    self.showErrorBannerWithMessage = self.applePayViewModel.validateCheckoutError
```

Comment:
It looks like this will prevent errors from showing unless they're apple card errors - Shouldn't we merge the `validateCheckoutExistingCardError` and `self.applePayViewModel.validateCheckoutError` signals instead?